Crates are mailed with manifests taped to the outside by paprka Merge Conflict resolution

astralenigma opened the pull request:
Resolves #7896 Merge conflicts
Adds a tear_manifest() to crates to remove copypasta.

astralenigma committed:
Added ignore to all thumbs I think

astralenigma committed:
Revert "Added ignore to all thumbs I think"

This reverts commit e358ada882e502a7b98df837f58b49505eb57485.

astralenigma committed:
Added Paprka's changelog

Since he's the true author

astralenigma committed:
The icons commit

astralenigma committed:
Putting all on github now.

Only crates is missing the overlay of the manifesto.

astralenigma committed:
The manifest code is completed in the crate

just need to fix possible bugs that I find.

astralenigma committed:
Second attempt at overlay

None of the crates appear with the manifesto outside, but he's there
waiting lurking in the shadows waiting to strike.

astralenigma committed:
Revert "The icons commit"

This reverts commit 7d0e93c15bafa0d849371b0e94260a70507badb0.

astralenigma committed:
This fixes the Icon problem now

Thank you Jordie

Jordie0608 commented on code/game/objects/structures/crates_lockers/crates.dm:
```diff
@@ -146,7 +151,7 @@
 /obj/structure/closet/crate/secure/New()
 	..()
-	overlays.Cut()
+	//overlays.Cut()
```
This can go.

Jordie0608 commented on code/game/objects/structures/crates_lockers/crates.dm:
```diff
@@ -10,16 +10,21 @@
+	
 /obj/structure/closet/crate/New()
 	..()
+	overlays.Cut()
```
This needs to be in update_icon()

astralenigma replied:
Should I also move the secure's crate addition of overlay to the update_icon()? or should I instead add the update_icon() to the new() before the addition of the overlay?

astralenigma replied:
Or should I override secure's update_icon()?

Cheridan commented:
:turtle:

pudl commented:
this guy's history is really fishy

Cheridan commented:
@pudl not as bad as your post history

astralenigma commented:
Damn here I thought my story was solid, where did I make a mistake? So that I can improve.

astralenigma commented on code/game/objects/structures/crates_lockers/largecrate.dm:
```diff
@@ -4,13 +4,38 @@
 	icon_state = "densecrate"
 	density = 1
+	var/obj/item/weapon/paper/manifest/manifest
+
```
The fact that largecrate is not a crate prevents the use of the tear_manifest() unless I drag tear_manifest() to structures but then you could tear manifests from every structure (maybe in the future cargo could order complete machines).

xxalpha commented:
Vou-me rir bastante se tu fores mesmo o pimento.

astralenigma commented:
Quem é o pimento?

PKPenguin321 commented:
"the pepper" is paprika who originally made this PR. paprika is banned from everything /tg/ because he ban evaded after getting banned for being rude.
when they say your history looks fishy they're saying it kinda looks like you're paprika ban evading due to your short history on git

astralenigma committed:
Moved the overlays.cut() per request.

astralenigma commented:
Ok that explains why paprika was not cleaning his own mess then, but no I just got kinda tired of waiting for cheridan and the code wasn't complicated to solve the conflicts.

Jordie0608 commented on code/game/objects/structures/crates_lockers/crates.dm:
```diff
@@ -146,7 +151,9 @@
 
 /obj/structure/closet/crate/secure/New()
 	..()
```
I should've mentioned I meant this entire statement can go; all it does is call parent.

astralenigma committed:
I think I'm done but I'm always wrong

Removed secure/new() due to it's current uselessness
Put all changes in the secure overlays inside the update icon
Found a bug where EMP act would only unlock the crate instead of
breaking it but would still apply the emag overlay I defaulted the fix
to breaking the crate.
Moved the variables from emag act and emp act upwards for compliance
with the update_icon() checks.

astralenigma commented on code/game/objects/structures/crates_lockers/crates.dm:
```diff
@@ -231,13 +250,12 @@
 
```
The method above says that when he gets card he only locks the crate.

optimumtact replied:
I believe the allowed function does the access checks for locking/unlocking

astralenigma replied:
I have tested it and it only locks I don't see what did I change that could make it happen like that so I think someone else made it like that either by mistake or on purpose, whichever reason will affect if I change it or not.

astralenigma commented on code/game/objects/structures/crates_lockers/crates.dm:
```diff
@@ -287,15 +305,14 @@
 			src.locked = 1
-			overlays.Cut()
-			overlays += redlight
+			update_icon()
```
So a emp_act Locks an unlocked crate?

optimumtact replied:
Yes, strange, but I think it's okay to leave it as is.

astralenigma commented:
I wonder if I should remove paprka as author and put myself.

optimumtact commented on html/changelogs/Paprka-crades.yml.txt:
```diff
@@ -0,0 +1,6 @@
+author: Paprka
```
Just set this to whatever you want if you want to change the change log author, i.e author: AstralEnigma

You cannot have two, but you could add something in changes -tweak: "Based on work by paprka"

astralenigma replied:
Honestly when I started it, it was mostly paprka's work now I'm not so sure due to all the lines I had to change to make the merge. I'm still thinking of making another change that will be moving my proc in crates to structures so I can also use it in the largecrate.

JJRcop replied:
You can have two..
